js/components/dateRange.js

```diff
@@ -0,0 +1,280 @@
+      type: String,
+      default: function() {
+        return '-- ' + T('widgets.form.dateRange.rangePlaceholder');
+      }
+    }
+  },
+  data: function() {
+    return {
+      dateRange: null,
+      isInputFocused: false,
+      isPreview: Fliplet.Env.get('preview'),
+      today: moment().locale('en').format('YYYY-MM-DD'),
+      selectedRange: ''
+    };
+  },
+  mounted: function() {
+    this.initDaterange();
+
+    if (this.defaultValueSource !== 'default') {
+      this.setValueFromDefaultSettings({
+        source: this.defaultValueSource,
+        key: this.defaultValueKey
+      });
+    }
+
+    if (this.autofill === 'default' || this.autofill === 'always') {
+      this.value = {
+        start: this.today,
+        end: this.today
+      };
+      this.empty = false;
+    }
+
+    if (this.autofill === 'empty') {
+      this.value = {
+        start: '',
+        end: ''
+      };
+
+      return;
+    }
+
+    this.$emit('_input', this.name, this.value, false, true);
+    this.$v.$reset();
+  },
+  validations: function() {
+    var rules = {
+      value: {}
+    };
+
+    if (this.required && !this.readonly) {
+      rules.value.required = window.validators.required;
+    }
+
+    return rules;
+  },
+  computed: {
+    isApplyCurrentDateField: function() {
+      return this.autofill === 'always' || this.autofill === 'default';
+    },
+    predefinedRanges: function() {
+      return [
+        {
+          label: T('widgets.form.dateRange.predefinedRanges.today')
+        },
+        {
+          label: T('widgets.form.dateRange.predefinedRanges.yesterday')
+        },
+        {
+          label: T('widgets.form.dateRange.predefinedRanges.tomorrow')
+        },
+        {
+          label: T('widgets.form.dateRange.predefinedRanges.nextWeek')
+        },
+        {
+          label: T('widgets.form.dateRange.predefinedRanges.lastWeek')
+        },
+        {
+          label: T('widgets.form.dateRange.predefinedRanges.nextMonth')
+        },
+        {
+          label: T('widgets.form.dateRange.predefinedRanges.lastMonth')
+        }
+      ];
+    }
+  },
+  watch: {
+    value: function(val) {
+      if (!val && ['default', 'always'].indexOf(this.autofill) > -1 && (this.required || this.autofill === 'always')) {
+        this.value = {
+          start: this.today,
+          end: this.today
+        };
+
+        if (this.dateRange) {
+          this.dateRange.set(val, true);
+        }
+
+        this.$emit('_input', this.name, val, false, true);
+
+        return;
+      }
+
+      if (this.isPreview && this.$v.value.$invalid) {
+        this.highlightError();
+      }
+
+      if (this.dateRange) {
+        this.dateRange.set(val, true);
+      }
+
+      this.$emit('_input', this.name, val, false, true);
+    },
+    selectedRange: function(val) {
+      var newDate = {
+        start: '',
+        end: ''
+      };
+
+      switch (val) {
+        case T('widgets.form.dateRange.predefinedRanges.today'):
```

Comment:
No need to check against localised string if you follow the `{ label, value }` structure I showed above.

You can then just use `newDate = this.getDate(val)`